src/clr-angular/forms/textarea/textarea.spec.ts

```diff
@@ -0,0 +1,94 @@
+import { ClrTextareaContainer } from './textarea-container';
+import { ClrCommonFormsModule } from '../common/common.module';
+import { WrappedFormControl } from '../common/wrapped-control';
+import { ControlIdService } from '../common/providers/control-id.service';
+
+@Component({
+  template: `
+    <textarea clrTextarea></textarea>
+    `,
+})
+class InvalidUseTest {}
+
+@Component({
+  template: `
+    <textarea clrTextarea name="model" [(ngModel)]="model"></textarea>
+    `,
+})
+class SimpleTest {}
+
+export default function(): void {
+  describe('Input directive', () => {
+    let textarea, fixture, ifErrorService, ngControlService;
+
+    describe('invalid use', () => {
+      it('throws error when used without a form control', () => {
+        TestBed.configureTestingModule({ declarations: [ClrTextarea, InvalidUseTest] });
+        expect(() => {
+          fixture = TestBed.createComponent(InvalidUseTest);
+          fixture.detectChanges();
+        }).toThrow();
+      });
+    });
+
+    describe('basic use', () => {
+      beforeEach(() => {
+        spyOn(WrappedFormControl.prototype, 'ngOnInit');
+        TestBed.configureTestingModule({
+          imports: [FormsModule, ClrIconModule, ClrCommonFormsModule],
+          declarations: [ClrTextarea, ClrTextareaContainer, SimpleTest],
+          providers: [IfErrorService, NgControlService, ControlIdService],
+        });
+        fixture = TestBed.createComponent(SimpleTest);
+        textarea = fixture.debugElement.query(By.directive(ClrTextarea));
+        ifErrorService = textarea.injector.get(IfErrorService, null);
+        ngControlService = textarea.injector.get(NgControlService, null);
+        spyOn(ifErrorService, 'triggerStatusChange');
+        spyOn(ngControlService, 'setControl');
+        fixture.detectChanges();
+      });
+
+      it('should apply the clr-textarea class', () => {
+        expect(textarea.nativeElement.classList.contains('clr-textarea'));
+      });
+
+      it('should have the IfErrorService', () => {
+        expect(ifErrorService).toBeTruthy();
+      });
+
+      it('should have the NgControlService and set the control', () => {
+        expect(ngControlService).toBeTruthy();
+        expect(ngControlService.setControl).toHaveBeenCalled();
+      });
+
+      it('correctly extends WrappedFormControl with ClrTextareaContainer', () => {
+        expect(textarea.injector.get(ClrTextarea).wrapperType).toBe(ClrTextareaContainer);
+        expect(WrappedFormControl.prototype.ngOnInit).toHaveBeenCalled();
+      });
+
+      it('should handle blur events', () => {
+        // textareas must be both invalid and blurred to register the validity
+        textarea.nativeElement.value = 'abc';
+        textarea.nativeElement.dispatchEvent(new Event('textarea'));
+        textarea.nativeElement.dispatchEvent(new Event('blur'));
+        fixture.detectChanges();
+        expect(ifErrorService.triggerStatusChange).toHaveBeenCalled();
+      });
+    });
```

Comment:
You're missing a test checking that it properly sets `controlClassService.className`.

Reply:
Added